### PR TITLE
Allow `setproperty!` for `PointerWrapper`s with special `struct`s

### PR DESCRIPTION
**Pull request by JoshuaLampert**
As discussed in #77 this adds the functionality to set properties also for structs as in #72 using the same approach like #74.

### CODE REVIEW

File: src/pointerwrappers.jl [hunk 1]
```diff
@@ -70,10 +70,12 @@ end
 function Base.setproperty!(pw::PointerWrapper{T}, name::Symbol, v) where T
   i = findfirst(isequal(name), fieldnames(T))
   if isnothing(i)
-    error("type $(string(T)) has no field $name")
+    # For some `struct`s, `fieldnames` gives `data` and not the actual field names, but we can use `Base.setproperty!` for pointers,
+    # see https://github.com/trixi-framework/P4est.jl/issues/72 and https://github.com/trixi-framework/P4est.jl/issues/79
+    Base.setproperty!(pointer(pw), name, v)
+  else
```
Comment (sloede) [inline src/pointerwrappers.jl:76]: Iirc, in the other Pr we didn’t have an if else condition but an if with an early return, or am I remembering this incorrectly? Either way you do it, it should imho be consistent between both functions.

**Review by sloede — CHANGES_REQUESTED** [review]
One small comment, otherwise this LGTM!

File: src/pointerwrappers.jl (hunk 1 above)
Comment (JoshuaLampert) [reply src/pointerwrappers.jl]: Done in 1483df31de8a9514610424d2dce3809ae9c53d4a.

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks! Could you please add a test of the new functionality?

**Comment by JoshuaLampert** [comment]
Do you think of an additional test than the ones I already added? If so, what kind of test were you thinking of?

**Review by ranocha — APPROVED** [review]
Thanks

**Pull request merged**